server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -267,4 +278,104 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
             }
         });
     }
+
+    void setRoutingWeights(
+        String awarenessAttributeName,
+        Map<String, Double> awarenessAttributeToWeightMap,
+        ActionListener<ClusterPutWeightedRoutingResponse> listener
+    ) {
+        // WRR API will validate invalid weights
+        final ClusterPutWeightedRoutingRequest clusterPutRoutingWeightRequest = new ClusterPutWeightedRoutingRequest(
+            awarenessAttributeName
+        );
+        clusterPutRoutingWeightRequest.setWeightedRouting(new WeightedRouting(awarenessAttributeName, awarenessAttributeToWeightMap));
+
+        transportService.sendRequest(
+            transportService.getLocalNode(),
+            ClusterAddWeightedRoutingAction.NAME,
+            clusterPutRoutingWeightRequest,
+            new TransportResponseHandler<ClusterPutWeightedRoutingResponse>() {
+                @Override
+                public void handleResponse(ClusterPutWeightedRoutingResponse response) {
+                    logger.info("Weights are successfully set. [{}]", clusterPutRoutingWeightRequest.getWeightedRouting().weights());
+                    listener.onResponse(response);
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    // Logging message on failure. Should we do Retry? If weights are not set should we fail?
+                    logger.error("Exception occurred while setting weights.Exception Messages - ", exp);
```

Comment:
`.Exception Messages` looks a little weird here
Maybe we can add the message like - 
`Failed to set routing weights [{}], weights and pass the exception

Reply:
Done.